tensorflow/java/src/main/native/tensor_jni.cc

```diff
@@ -433,3 +472,15 @@ JNIEXPORT void JNICALL Java_org_tensorflow_Tensor_readNDArray(JNIEnv* env,
   readNDArray(env, dtype, static_cast<const char*>(data), sz, num_dims,
               static_cast<jarray>(value));
 }
+
+JNIEXPORT jobject JNICALL Java_org_tensorflow_Tensor_buffer(JNIEnv* env,
+                                                              jclass clazz,
+                                                              jlong handle) {
+  TF_Tensor* t = requireHandle(env, handle);
+  if (t == nullptr) return nullptr;
+  void* data = TF_TensorData(t);
+  const size_t sz = TF_TensorByteSize(t);
+
+  jobject dataBuffer = env->NewDirectByteBuffer(data, (jlong) sz);
```

Comment:
Nits:
- Use `static_cast<jlong>` instead of c-style casts
- Perhaps just `return env->NewDirectByteBuffer(...)`